CMakeLists.txt

```diff
@@ -2100,6 +2100,91 @@ list(APPEND
   ${extra_post_build_byproducts}
   )
 
+get_property(mcuboot_bin_byproducts
+  GLOBAL PROPERTY
+  mcuboot_bin_byproducts
+  )
+
+get_property(mcuboot_slot1_bin_byproducts
+  GLOBAL PROPERTY
+  mcuboot_slot1_bin_byproducts
+  )
+
+# Pack the mcuboot-signed .bin files into UF2 format
+if(CONFIG_BUILD_OUTPUT_UF2 AND CONFIG_BOOTLOADER_MCUBOOT)
```

Comment:
we should not do mcuboot image post-processing at toplevel CMakeLists.txt file.

Let `mcuboot.cmake` do what it needs to do for signing and placing the signed image in correct file type.

What I believe @nordicjm objected to in this comment: https://github.com/zephyrproject-rtos/zephyr/pull/89015#discussion_r2057967445 was that the implementation of a generic `convert_to_uf2()` was placed in `mcuboot.cmake`.

He was not against mcuboot.cmake being able to output a signed image in uf2 format by either place it there directly as part of signing process (like it's done for bin/hex files) or by converting an already signed bin file to uf2 format, for example by using a `convert_to_uf2()` function.